app/js/FactorialService.js

```diff
@@ -1,11 +1,11 @@
 var FactorialService = (function () {
 
     var _results = [1, 1];
-
+      
+    var _reg = new RegExp('^[0-9]+$');   
+     
     function _isNumberValid(number) {
-        return (number !== undefined)
-            && (number > -1)
-            && (typeof(number) !== 'string');
```

Comment:
We need this validation to make sure we are not dealing with a number inside into a string